transaction/merkletreeparent.go

```diff
@@ -0,0 +1,54 @@
+package transaction
+
+import (
+	"encoding/hex"
+
+	"github.com/bitcoin-sv/go-sdk/chainhash"
+	crypto "github.com/bitcoin-sv/go-sdk/primitives/hash"
+	"github.com/bitcoin-sv/go-sdk/util"
+)
+
+// MerkleTreeParentStr returns the Merkle Tree parent of two Merkle
+// Tree children using hex strings instead of just bytes.
+func MerkleTreeParentStr(leftNode, rightNode string) (string, error) {
+	l, err := hex.DecodeString(leftNode)
+	if err != nil {
+		return "", err
+	}
+	r, err := hex.DecodeString(rightNode)
+	if err != nil {
+		return "", err
+	}
+
+	return hex.EncodeToString(MerkleTreeParent(l, r)), nil
+}
+
+// MerkleTreeParent returns the Merkle Tree parent of two Merkle
+// Tree children.
+func MerkleTreeParent(leftNode, rightNode []byte) []byte {
+	// swap endianness before concatenating
+	l := util.ReverseBytes(leftNode)
+	r := util.ReverseBytes(rightNode)
```

Comment:
Consider my suggesion I gave you when you've been updating the `go-paymail` to use `go-sdk`.

https://github.com/bitcoin-sv/go-paymail/pull/110#discussion_r1766197764

https://go.dev/play/p/lTznDklvMCo

Reply:
After internal discussion I hope that I addressed all performance upgrades :D